libs/auth-api-lib/src/lib/services/resources.service.ts

```diff
@@ -322,15 +325,20 @@ export class ResourcesService {
     return await this.apiScopeModel.destroy({ where: { name: name } })
   }
 
-  /** Deletes an API resource */
-  async deleteApiResource(name: string): Promise<number> {
-    this.logger.debug('Deleting api resource with name: ', name)
+  /** Archives an API resource */
+  async archiveApiResource(name: string): Promise<number> {
+    this.logger.debug('Archiving api resource with name: ', name)
 
     if (!name) {
       throw new BadRequestException('Name must be provided')
     }
 
-    return await this.apiResourceModel.destroy({ where: { name: name } })
+    const result = await this.apiResourceModel.update(
+      { archived: new Date() },
+      { where: { name: name } },
+    )
+
+    return result[0]
   }
 
   async getResourceUserClaims(name: string): Promise<any> {
```

Comment:
What about the Models and DTO's? Probably not needed in DTO's in this implementation, but isn't this field suppose to at least be in the models?

Reply:
Can´t figure out how to let sequelize.query return a model. Default it´s unknown. 
Set it as todo for now